### ISSUE TITLE
Add support for custom conditions on task groups

### ISSUE THREAD

**Issue by kzu** [microsoft/azure-pipelines-tasks#5112]
Currently, individual tasks can be assigned a [custom condition](https://www.visualstudio.com/en-us/docs/build/concepts/process/conditions). Task groups don't have that capability though.

**Comment by bryanmacfarlane** [comment]
Yeah, it's in the works.  It's not a task issue but a service side issue.  task group support is coming soon (the main issue is the task group UI is still using the old editor)

**Comment by rmarinho** [comment]
👍 on this.. any news @bryanmacfarlane ?

**Comment by ivan-prodanov** [comment]
@rmarinho I just came to the conclusion that I need this and I need it now. After observing @bryanmacfarlane's comment I realized that if we set the condition to a task and then create a task group from that task, the condition will still be set, just not visible until they deploy the new UI. I just tried it and it works. 🥇

**Comment by kzu** [comment]
Yep, indeed. I found this out by coincidence.

**Comment by frankfuu** [comment]
Any ETA for this release?

**Comment by sibbl** [comment]
If someone else is looking for a workaround: using the dev chrome tools, search for `.prototype.beginUpdateTaskGroup` in MS's code. In there, the variable `t` contains the JSON object sent to the REST endpoint. You can set a breakpoint beforehand and modify it as required.

**Comment by bryanmacfarlane** [comment]
@jstawski - see the comment that went along with closing it.

**Comment by stijnherreman** [comment]
@bryanmacfarlane Is there a publicly tracked issue for migrating Task Groups to the new UI?

**Comment by jstawski** [comment]
I'm not sure if this merits a new bug or not, but here some feedback. As of today, it seems the only way to add a variable into a task group is to reference one in one of the tasks. That is problematic if the variable is only used on the custom condition as it is not being checked. I had to resolve by using a dummy task and reference the variable there, but it truly shouldn't be like that. Either add support to check for new variables on the custom condition or the ability to manually add a variable.

For others that might run into this problem:
My dummy condition is a `Command Line` task:
Tool: `echo`
Arguments: `"$(VariableName)"`

**Comment by deka** [comment]
Yes, not usable if the variable is only used on the custom condition.

**Comment by bergmeister** [comment]
@bryanmacfarlane This still does not seem to be implemented

**Comment by bryanmacfarlane** [comment]
@bansalaseem for task group question.

**Comment by bryanmacfarlane** [comment]
Since this is a service side issue, developer community would be a better channel.  That's the only publicly tracking mechanism for the service.

**Comment by bansalaseem** [comment]
Thanks @jstawski for the feedback. Right now the workaround which you have done looks reasonable and we have a story in our backlog to manually specify a task group variable. I dont have a specific timeline when we will get to it though.

**Comment by AroglDarthu** [comment]
> Yeah, it's in the works. It's not a task issue but a service side issue. task group support is coming soon (the main issue is the task group UI is still using the old editor)

We are using TFS 2018 Update 2. There is still no option to add a condition to the Task Group!
I want to NOT RUN the entire task group, based on a single condition. Seems overkill to introduce a variable in one of the underlying tasks and then add a custom run condition on each of the tasks with the group. That's just really silly!

So why is this issue closed? There is a valid use case for having a custom run condition on the task group itself.

**Comment by harry12311** [comment]
Is there an update on this? I have the same requirement as AroglDarthu. I am using Azure DevOps and see no option to control execution of task group apart from hacky solution!

**Comment by guillermomartinez-accenture** [comment]
Are you planning to enable Custom Conditions on task groups any time soon?
The initial request was from Aug 2017.

**Comment by joshbrw** [comment]
Came here looking for this, almost 2 years later. Any updates?

**Comment by joshbrw** [comment]
Following on from my message above; the way that I ended up solving this was using separate Agent Phases and putting the task group into an Agent Phase that I've set up with execution conditions. If you do use this path, make sure you set the `Dependencies` on each phase so that it only runs if previous phases have been completed.

**Comment by ashokirla** [comment]
Due to few other priority items like pipelines YAML, we couldn't find bandwidth to work on this feature. As of now the backlog for the next 2 months is full. We will try to pick up task group conditions feature after that. Will keep you posted.

**Comment by robal** [comment]
Please add this. I need this to be able to run Task Groups in parallel using "multi-configuration".
For the impatient ones: I've found a sick workaround:

1) take your Task Group and add a dummy task to print two vars (this is just to make the vars appear on Task Group variable list).  Eg:  
Inline Powershell: `Write-Host("$(selector) $(staticselector)")`

2) specify Custom Condition on ALL tasks in the Task Group to:
 `and(succeeded(), eq(variables['selector'], variables['staticselector']))`

![image](https://user-images.githubusercontent.com/5574519/66052838-2cd3ab00-e529-11e9-862e-542889176de0.png)


3) Add `$selector` variable to your release, set to a multi-configuration CSV string, eg:  `"thread_one,thread_two,thread_three"`

3) add your Task Group three times to a Multi-Configuration Agent Job. For each one, set:
`$selector` to `$(selector)`
`$staticselector` to one of the CSV values, eg `"thread_one"` for the first instance, `"thread_two"` for the second and `"thread_three"` for the third

![image](https://user-images.githubusercontent.com/5574519/66053046-8b008e00-e529-11e9-9342-0ac2df209b7c.png)

4) Profit!  Each multi-configuration job runs a separate thread of your Task Group with thread-specific parameters

![image](https://user-images.githubusercontent.com/5574519/66053174-bc795980-e529-11e9-8a00-1d2ae4ef151a.png)

**Comment by paul-roberts-dt** [comment]
Another vote for this.

**Comment by shayki5** [comment]
+1 !

**Comment by maxkoshevoi** [comment]
+1

**Comment by anatolybolshakov** [comment]
Attached [link](https://developercommunity.visualstudio.com/idea/365689/task-group-custom-conditions-at-group-and-task-lev.html) to related Developer community issue - since it would be better place to get last updates about it's status.

**Comment by juliank** [comment]
@anatolybolshakov: Are you or anybody else from the team able to get some more information about this issue? It is 5 years since this issue was posted and two years since the last comment. In the related [Developer community issue](https://developercommunity.visualstudio.com/t/task-group-custom-conditions-at-group-and-task-lev/365689) you link to there has been no official answer/activity for years 😞

**Comment by mhoolahan** [comment]
typical microsoft product tbh. introduce a new feature that's somewhat useful. spend the extra sprint or two to make it really useful? no sorry...

**Comment by pdsrebelo** [comment]
+1 on this please, it has been almost 6 years now since we asked for this simple feature.

**Comment by nickbrooking** [comment]
Adding another request for this. I'm not getting my hopes up but man it would be nice to have this, we have to do some janky workarounds until this exists.

**Comment by SimmeNilsson** [comment]
+1

**Comment by FedericoDesio** [comment]
+1

**Comment by wsmigielskialp** [comment]
+1 in 2025, quite upset this isn't a feature and is treated as low priority when it would improves probably thousands of pipelines...
